pinot-hadoop/src/main/java/org/apache/pinot/hadoop/job/SegmentCreationJob.java

```diff
@@ -194,14 +193,6 @@ public void run()
     _fileSystem.delete(_stagingDir, true);
   }
 
-  @Nullable
-  protected TableConfig getTableConfig()
-      throws IOException {
-    try (ControllerRestApi controllerRestApi = getControllerRestApi()) {
-      return controllerRestApi != null ? controllerRestApi.getTableConfig() : null;
-    }
-  }
-
   protected Schema getSchema()
```

Comment:
Move this to Base class as well

Reply:
The way we support schemas means we potentially need to initialize the filesystem and open the schema file. Without running the job (as this is an abstract class), we cannot do that.

Reply:
Actually we can move _filesystem and _schemaFile into the base class as well. Your call